modules/Media.js

```diff
@@ -22,37 +29,68 @@ class Media extends React.Component {
     matches: true
   }
 
-  updateMatches = () =>
-    this.setState({ matches: this.mediaQueryList.matches })
+  updateMatches = () => {
+    let { query, queries } = this.props
+    if (query)
+      this.setState({ matches: this.mediaQueryList.matches })
+
+    if (queries)
+      this.setState({
+        matches: this.mediaQueryList.reduce((accumulated, { name, mm }) => ({
+          ...accumulated,
+          [name]: mm.matches,
+        }), {}),
+      })
+  }
 
   componentWillMount() {
     if (typeof window !== 'object')
       return
 
-    let { query } = this.props
+    let { query, queries } = this.props
 
-    if (typeof query !== 'string')
+    if (query && typeof query !== 'string')
       query = json2mq(query)
 
-    this.mediaQueryList = window.matchMedia(query)
-    this.mediaQueryList.addListener(this.updateMatches)
+    if (query) {
+      this.mediaQueryList = window.matchMedia(query)
+      this.mediaQueryList.addListener(this.updateMatches)
+    }
+
+    if (queries) {
+      queries = Object.keys(queries).map(mq => ({
+        name: mq, 
+        qs: json2mq(queries[mq]),
+      }))
+      this.mediaQueryList = queries.map(mq => ({
```

Comment:
In the case of a single query, `this.mediaQueryList` is an actual [`MediaQueryList`](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList) object. In the case of multiple queries, it's an array. I'd prefer it to be consistent.